static/panes/opt-pipeline.ts

```diff
@@ -342,6 +344,11 @@ export class OptPipeline extends MonacoPane<monaco.editor.IStandaloneDiffEditor,
         if (compiler && !compiler.optPipeline) {
             //this.editor.setValue('<Opt pipeline output is not supported for this compiler>');
         }
+
+        // TODO: un-hackify this
+        if (this.compilerInfo.compilerName.includes('clangir')) {
+            this.options.enableToggle('dump-full-module', false, true);
+        }
```

Comment:
Sorry..

Reply:
Would it be better to disable the button in the front-end and enforce dump full module on the back-end for clang-ir?

Reply:
I agree but that would take some refactoring. 
It seems like the right way to communicate this to the front-end would be through `compiler.optPipeline`, maybe instead of:
```
        compiler.compiler.optPipeline = {
...
            supportedOptions: ['dump-full-module', 'demangle-symbols'],
        };
```
do
```
        compiler.compiler.optPipeline = {
...
            supportedOptions: ['dump-full-module': 'on', 'demangle-symbols': 'free'],
        };
```
And adjust `OptPipeline.updateButtons` to parse and act accordingly?  That wouldn't be backwards compatible of course, it would require to modify and test all other parsers that use supportedOptions/Filters.  WDYT?

Reply:
Scratch that, I think I know how to do it properly